### ISSUE TITLE
How to set legend item labels for the markers?

### ISSUE THREAD

**Issue by mzhukovs** [rasbt/mlxtend#452]
In all examples I see 0, 1, etc. numbers for the labels in the legend - is there a way to set those labels to specified strings? 

I tried using scatter_kwargs with a dictionary that contained a key for "label" (like in the underlying matplotlib scatter function) but didn't work.

Would appreciate the tip, or an enhancement if currently not possible.

Thanks for the library.

**Comment by mzhukovs** [comment]
Ok figured it out but it's ugly (new to python so please let me know if there's a better way, or if you believe it's worth adding it as an enhancement to your library to make it easier if not). The other ways I tried, e.g. the more direct fig.legend(labels=markerLabels), would show the label text as desired but would screw up the plot (somehow an orange box appeared that covered up things randomly) and removed the markers from the legend.

relevant snippets:

```
markerLabels = ['Red Wine', 'White Wine']
legendLabels = list(zip(markerLabels, np.arange(len(markerLabels))))

    fig = plot_decision_regions(X=X, y=y.astype(np.integer), clf=clf, legend=2)
    legendText = fig.get_legend().get_texts()
    [legendText[legendLabel[1]].set_text(legendLabel[0]) for legendLabel in legendLabels]
```

**Comment by rasbt** [comment]
Hi there,

something simple like `plt.legend([...])` used to work, but after we added several extensions/options to the `plot_decision_regions` function, this does not seem to work properly anymore.

A solution would be the following:

```python
handles, labels = ax.get_legend_handles_labels()
ax.legend(handles, 
          ['class 0', 'class 1', 'class 2'], 
           framealpha=0.3, scatterpoints=1)
```

A full example is shown below

```python
from mlxtend.plotting import plot_decision_regions
import matplotlib.pyplot as plt
from sklearn import datasets
from sklearn.svm import SVC

# Loading some example data
iris = datasets.load_iris()
X = iris.data[:, [0, 2]]
y = iris.target

# Training a classifier
svm = SVC(C=0.5, kernel='linear')
svm.fit(X, y)


# Plotting decision regions
ax = plot_decision_regions(X, y, clf=svm, legend=0)

# Adding axes annotations
plt.xlabel('sepal length [cm]')
plt.ylabel('petal length [cm]')
plt.title('SVM on Iris')

handles, labels = ax.get_legend_handles_labels()
ax.legend(handles, 
          ['class 0', 'class 1', 'class 2'], 
           framealpha=0.3, scatterpoints=1)

plt.show()
```

![unknown](https://user-images.githubusercontent.com/5618407/47258853-2ad32200-d467-11e8-8b5e-e1c437d1849b.png)

Hope this solves your problem. Also, I will add this to the documentation
